openfl/cryptography/ca.py

```diff
@@ -112,6 +112,10 @@ def sign_certificate(csr, issuer_private_key, issuer_name, days_to_expiration=36
     builder = builder.add_extension(
         x509.BasicConstraints(ca=ca, path_length=None), critical=True,
     )
+    try:
+        builder = builder.add_extension(csr.extensions.get_extension_for_oid(ExtensionOID.SUBJECT_ALTERNATIVE_NAME).value, critical=False)
+    except ExtensionNotFound as e:
+        pass  # Might not have alternative name
```

Comment:
This is the only change required for IP-based communication to work.

Reply:
The other's are to ensure that the intermediate output during plan generation actually honor the FQDN specified by the user, to reduce confusion when debugging